internal/service/certificatesigningrequest.go

```diff
@@ -274,9 +274,17 @@ func (h *ServiceHandler) ApproveCertificateSigningRequest(ctx context.Context, r
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been approved"}, nil
 	}
 
-	signedCert, err := signApprovedCertificateSigningRequest(h.ca, csr)
+	signedCert, err, errctx := signApprovedCertificateSigningRequest(h.ca, csr)
 	if err != nil {
-		return nil, err
+		switch err {
+		case flterrors.ErrCNLength:
+			msg := fmt.Sprintf("cn: %s does not meet requirement: %s", errctx, err.Error())
+			return server.ApproveCertificateSigningRequest422JSONResponse{Message: msg}, nil
+		case flterrors.ErrInvalidPEMBlock:
+			return server.ApproveCertificateSigningRequest422JSONResponse{Message: err.Error()}, nil
+		default:
+			return server.ApproveCertificateSigningRequest500JSONResponse{Message: err.Error()}, nil
```

Comment:
lets use `errors.Is`
```suggestion
		switch {
		case errors.Is(err, flterrors.ErrCNLength) || errors.Is(err, flterrors.ErrInvalidPEMBlock):
			return server.ApproveCertificateSigningRequest422JSONResponse{Message: err.Error()}, nil
		default:
			return server.ApproveCertificateSigningRequest500JSONResponse{Message: err.Error()}, nil
		}
```